### PR TITLE
[DPEDE-895] Create event that returns clicked element in autocomplete dropdown menu

### PR DESCRIPTION
**Pull request by ASanchLumen**
https://ctl.atlassian.net/browse/DPEDE-895

- Adds `chiSearchInputItemSelected` event in search-input wc that emits the item clicked
- Adds property `disableItemSelectedEvent` in dropdown wc that disables that event, as it is not working properly when this component is used in search-input wc
- Sets `disable-item-selected-event` in the dropdown used in searc-input wc.

### CODE REVIEW

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -63,6 +63,10 @@ export class Dropdown {
    * To prevent hiding of the Dropdown when clicking outside its bounds
    */
   @Prop({ reflect: true }) preventAutoHide: boolean;
+  /**
+   * To prevent emitting the event chiDropdownItemSelected when an item is clicked
+   */
+  @Prop() disableItemSelectedEvent = false;
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:69]:
```suggestion
  @Prop() preventItemSelectedEvent = false;
```

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 2]
```diff
@@ -320,18 +323,24 @@ export class Dropdown {
...
 
     document.body.addEventListener('click', this.handlerClick);
-    menuItems.forEach((item: HTMLElement) => {
-      item.addEventListener('click', this.handlerSelectedMenuItem);
-    });
+
+    if (!this.disableItemSelectedEvent) {
+      menuItems.forEach((item: HTMLElement) => {
+        item.addEventListener('click', this.handlerSelectedMenuItem);
+      });
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:330]: Same in `_removeEventListeners()`

```suggestion
    if (this.preventItemSelectedEvent) return;
    
    menuItems.forEach((item: HTMLElement) => {
       item.addEventListener('click', this.handlerSelectedMenuItem);
    });
```

File: src/custom-elements/src/components/search-input/search-input.tsx [hunk 3]
```diff
@@ -84,7 +84,11 @@ export class SearchInput {
   /**
    * Triggered when the user clicked the Search button.
    */
-  @Event({ eventName: 'chiSearch' }) eventSearch: EventEmitter;
+  @Event({ eventName: 'chiSearch' }) eventSearch: EventEmitter<string>;
+  /**
+   * Triggered when the user clicked in one of the autocomplete elements
+   */
+  @Event({ eventName: `chiSearchInputItemSelected`}) itemSelected: EventEmitter<DropdownMenuItem>
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/search-input/search-input.tsx:91]: Single quotes. Is this name correct: `chiSearchInputItemSelected`? @jllr 

```suggestion
  @Event({ eventName: 'chiSearchInputItemSelected'}) eventItemSelected: EventEmitter<DropdownMenuItem>
```

File: src/custom-elements/src/components/search-input/search-input.tsx [hunk 4]
```diff
@@ -264,7 +245,7 @@ export class SearchInput {
       <chi-dropdown
         id="dropdown-autocomplete"
         position="bottom"
-        prevent-auto-hide
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/search-input/search-input.tsx:267]: Good catch!

File: test/chi/custom-elements/search-input.pug [hunk 5]
```diff
@@ -38,6 +38,8 @@ script.
       { title: "Item 2", href: "#" },
       { title: "Item 3", href: "#" },
       { title: "Item 4", href: "#" },
-      { title: "Item 5", href: "#" }
+      { title: "Item 5", href: "#"}
     ];
   });
+  document.addEventListener('chiSearchInputItemSelected', (ev) => console.log("chiItemSelected", ev.detail));
+  document.addEventListener('chiDropdownItemSelected', (ev) => console.log("chiDropdownItemSelected", ev.detail));
```
Comment (josecarlosgonzalezv) [inline test/chi/custom-elements/search-input.pug:45]: I don't think it's necessary to include listeners here.

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 6]
```diff
@@ -82,8 +86,7 @@ export class Dropdown {
   /**
    * Triggered when select an item in the dropdown menu
    */
-  @Event({ eventName: 'chiDropdownItemSelected' })
-  eventItemSelected: EventEmitter;
+  @Event({ eventName: 'chiDropdownItemSelected' }) eventItemSelected: EventEmitter;
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:89]: Be careful with linting, I would remove all the linting that you have changed.

**Review by josecarlosgonzalezv — CHANGES_REQUESTED** [review]

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 7]
```diff
@@ -320,6 +324,9 @@ export class Dropdown {
     const menuItems = this._getDropdownMenuItems();
 
     document.body.addEventListener('click', this.handlerClick);
+
+    if (this.preventItemSelectedEvent) return
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:328]:
```suggestion
    if (this.preventItemSelectedEvent) return;
```

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 8]
```diff
@@ -329,6 +336,9 @@ export class Dropdown {
     const menuItems = this._getDropdownMenuItems();
 
     document.body.removeEventListener('click', this.handlerClick);
+
+    if (this.preventItemSelectedEvent) return
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:340]:
```suggestion
    if (this.preventItemSelectedEvent) return;
```

File: test/chi/custom-elements/search-input.pug [hunk 9]
```diff
@@ -38,6 +38,6 @@ script.
       { title: "Item 2", href: "#" },
       { title: "Item 3", href: "#" },
       { title: "Item 4", href: "#" },
-      { title: "Item 5", href: "#" }
+      { title: "Item 5", href: "#"}
```
Comment (josecarlosgonzalezv) [inline test/chi/custom-elements/search-input.pug:41]:
```suggestion
      { title: "Item 5", href: "#" }
```

**Review by josecarlosgonzalezv — APPROVED** [review]

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 10]
```diff
@@ -63,6 +63,10 @@ export class Dropdown {
    * To prevent hiding of the Dropdown when clicking outside its bounds
    */
   @Prop({ reflect: true }) preventAutoHide: boolean;
+  /**
+   * To prevent emitting the event chiDropdownItemSelected when an item is clicked
+   */
+  @Prop() preventItemSelectedEvent = false;
```
Comment (jllr) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:69]:
```suggestion
  @Prop() preventItemSelected = false;
```

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 11]
```diff
@@ -320,6 +324,9 @@ export class Dropdown {
     const menuItems = this._getDropdownMenuItems();
 
     document.body.addEventListener('click', this.handlerClick);
+
+    if (this.preventItemSelectedEvent) return;
```
Comment (jllr) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:328]:
```suggestion
    if (this.preventItemSelected) return;
```

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 12]
```diff
@@ -329,6 +336,9 @@ export class Dropdown {
     const menuItems = this._getDropdownMenuItems();
 
     document.body.removeEventListener('click', this.handlerClick);
+
+    if (this.preventItemSelectedEvent) return;
```
Comment (jllr) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:340]:
```suggestion
    if (this.preventItemSelected) return;
```

File: src/custom-elements/src/components/search-input/search-input.tsx [hunk 13]
```diff
@@ -264,7 +245,7 @@ export class SearchInput {
...
+        disable-item-selected-event
```
Comment (jllr) [inline src/custom-elements/src/components/search-input/search-input.tsx:248]:
```suggestion
        prevent-item-selected
```

**Review by jllr — APPROVED** [review]

**Pull request merged**